server/cli/launch.py

```diff
@@ -183,4 +188,9 @@ def launch(
         f = open(devnull, "w")
         sys.stdout = f
 
-    server.app.run(host=host, debug=debug, port=port, threaded=True)
+    try:
+        server.app.run(host=host, debug=debug, port=port, threaded=True)
+    except OSError as e:
+        if e.errno == errno.EADDRINUSE:
+            raise click.ClickException(f"{e} Port is in use, please specify an open port using the --port flag.")
+        raise click.ClickException(f"{e}")
```

Comment:
I'm not sure why you are raising click.ClickException rather than just reraising the original exception (just `raise` will do that).

Reply:
good catch, this isn't an expected error so I don't want click to handle it. will fix